bertie_bogo_shim/src/main.rs

```diff
@@ -0,0 +1,214 @@
+// # Bertie BoGo Shim
+//
+// # Credit
+//
+// The code is based on rustls' bogo shim ...
+//
+//     https://github.com/rustls/rustls/blob/main/rustls/examples/internal/bogo_shim.rs
+//
+
+extern crate core;
+
+use std::{env, net::TcpStream, process};
+use tracing::Level;
+
+use simple_https_client::tls13client;
+
+static BOGO_NACK: i32 = 89;
+
+#[derive(Debug)]
+struct Options {
+    port: u16,
+    role: Role,
+    key_file: String,
+    cert_file: String,
+    hostname: String,
+    expect_fallback_scsv: bool,
+    expect_extended_master_secret: bool,
+}
+
+impl Default for Options {
```

Comment:
This could be derived if you implement `Default` for `Role`. But it's a little weird to have `example.com` in here as default.
Not too important though.